### ISSUE TITLE
Calculating Log Loss for Negative and Positive Values

### ISSUE THREAD

**Issue by skarakulak** [keras-team/keras#9713]
Hi, 

I am intending to use a log-loss function that would work both in negative and positive values. Obviously, the log function is not defined for negative values, so the formula I want to use is as below:

```
if (x >= 0):
    log(1+x)
else:
    (-1) * (log(1+(-1)*x))
```

Do you have any suggestion as to how to update the log-loss function below to archive this.
```
def mean_squared_logarithmic_error(y_true, y_pred):
    first_log = K.log(K.clip(y_pred, K.epsilon(), None) + 1.)
    second_log = K.log(K.clip(y_true, K.epsilon(), None) + 1.)
return K.mean(K.square(first_log - second_log), axis=-1)
```

Thanks in advance:)

**Comment by skarakulak** [comment]
I guess this will do it:) 

```
def mean_squared_logarithmic_error_with_negs(y_true, y_pred):
    first_log =   K.log(K.clip(y_pred, K.epsilon(), None) + 1.) + (-1) * K.log( K.abs(K.clip(y_pred, None, (-1) * K.epsilon())) + 1.)
    second_log  = K.log(K.clip(y_true, K.epsilon(), None) + 1.) + (-1) * K.log( K.abs(K.clip(y_true, None, (-1) * K.epsilon())) + 1.)
    return K.mean(K.square(first_log - second_log), axis=-1)
```

**Comment by tRosenflanz** [comment]
Why not `K.sign(x)*K.log(1+K.abs(x))`

**Comment by skarakulak** [comment]
That's also neat, thank you:)

By the way, is this the correct way to apply class weights? The original code is a bit different, but let's assume the example below demonstrates my suggestions for an output with 5 classes, where the first output is has the weight of 5.
```

def mean_squared_logarithmic_error_with_negs(y_true, y_pred):
    weigts_c = np.ones(5)
    weigts_c[0]=5
    first_log =   K.sign(x)*K.log(1+K.abs(y_pred))
    second_log  = K.sign(x)*K.log(1+K.abs(y_true))
    return (K.mean(K.square(first_log - second_log)* weigts_c, axis=-1))
```
